Add README for benchmark module and benchmark sample application (#2167)
DataDog/dd-sdk-android: tools/benchmark/README.md

# Datadog benchmark module

## Overview

> This is a module for Datadog internal use to benchmarking the overhead of SDK features.

Reviewer: ```suggestion
This is a module for Datadog internal use to benchmark the overhead of SDK features.
```

> Currently the gauge that it measures includes CPU ticks, memory usage, and FPS of the application.

Reviewer 2: ```suggestion
Currently the gauges that it measures include CPU ticks, memory usage, and FPS of the application.
```

> Currently the gauges that it measures include CPU ticks, memory usage, and FPS of the application.

Reviewer: ```suggestion
It measures gauges that include CPU ticks, memory usage, and FPS of the application.
```

> Currently the gauge that it measures includes CPU ticks, memory usage and FPS of the application.

Reviewer: ```suggestion
Currently the gauge that it measures includes CPU ticks, memory usage, and FPS of the application.
```

## Getting started

> 1. Build Datadog exporter configuration

Reviewer: ```suggestion
1. Build the Datadog exporter configuration:
```

```kotlin
val configuration = DatadogExporterConfiguration.Builder("API_KEY")
    .setApplicationId("YOUR_APPLICATION_ID")
    .setApplicationName("Benchmark Application")
    .setApplicationVersion(BuildConfig.VERSION_NAME)
    .setIntervalInSeconds(30L)
    .build()

```

> 2. Create `DatadogMeter` with configuration

Reviewer: ```suggestion
2. Create `DatadogMeter` with the configuration:
```

``` kotlin
val meter = DatadogMeter.create(configuration)
```

> 3. Start all gauges with meter

Reviewer: ```suggestion
3. Start all gauges with meter:
```

```kotlin
meter.startGauges()
```

> 4. Stop all gauges with meter

Reviewer: ```suggestion
4. Stop all gauges with meter:
```

```kotlin
meter.stopGauges()
```

## Observe Metrics

All the metrics are uploaded

> through [Datadog submit metrics API](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics),

Reviewer: ```suggestion
through [Datadog submit metrics API](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics).
```

> The metrics name for each gauge:

Reviewer 2: ```suggestion
The metric names for each gauge:
```

* [android.benchmark.cpu]: CPU ticks
* [android.benchmark.memory]: Memory usage

> * [android.benchmark.fps]: Frame per second

Reviewer 2: ```suggestion
* [android.benchmark.fps]: Frames per second
```
